File: frontend/src/objects/review/ReviewAppealReview.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AppealStatus,
  IsoDateTime,
  ResolutionText,
} from '@/objects/core/SharedObjects'

export type ReviewAppealReview = {
  status: AppealStatus
  resolutionNote?: ResolutionText
  submittedAt: IsoDateTime
  reviewedAt?: IsoDateTime
}

File: backend/src/main/scala/services/review/objects/ReviewAppealReview.scala
package services.review.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class ReviewAppealReview(
    status: AppealStatus,
    resolutionNote: Option[ResolutionText],
    submittedAt: IsoDateTime,
    reviewedAt: Option[IsoDateTime],
)
object ReviewAppealReview:
  given Encoder[ReviewAppealReview] = deriveEncoder
  given Decoder[ReviewAppealReview] = deriveDecoder
